src/Microsoft.PowerShell.Commands.Management/commands/management/NewPropertyCommand.cs

```diff
@@ -266,16 +266,12 @@ private static bool IsRegistryProvider(IDictionary fakeBoundParameters)
         /// <param name="isLiteralPath">Specifies if path is literal path.</param>
         /// <returns>Collection of Pathinfo objects.</returns>
         private static Collection<PathInfo> ResolvePath(object path, bool isLiteralPath)
-        {
-            using var ps = System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace);
-
-            ps.AddCommand("Microsoft.PowerShell.Management\\Resolve-Path");
-            ps.AddParameter(isLiteralPath ? "LiteralPath" : "Path", path);
-
-            Collection<PathInfo> output = ps.Invoke<PathInfo>();
-
-            return output;
-        }
+            => CompletionCompleters.InvokeCommand<PathInfo>(
+                commandName: "Resolve-Path",
+                commandParameters: new Dictionary<string, object>
+                {
+                    { isLiteralPath ? "LiteralPath" : "Path", path },
+                });
```

Comment:
I wonder how this is related to completion code? It seems unrelated change. I see no value to generalize GetCommandInfo.

Reply:
Hmm do you mean in general this has no value? Tbh there're many places which are invoking a command in completer so to me it makes sense to generalise to a method.

But yeah if you see no value I can close the PR 🙂.

Reply:
I suggest to convert `InvokeCommand` to `InvokeGetCommand` and use it only for running Get-Command and only for completors.
(Ideally I'd see CompletionCompleters out of Engine but still in SDK. It is only thoughts, not a request.)

Reply:
Yeah so reason I did not do that was we have many commands being called e.g. Resolve-Path, Get-Command, Get-ExperimentalFeature etc. which is are invoked the same way using `ps.Invoke<T>`. They can return anything, not just `CommandInfo`.

So in this case writing a generic `InvokeCommand<T>` seems much more suitable.

I guess we could just split all into separate methods like `InvokeGetCommand` and `InvokeResolvePath`.

I would prefer either of these, but what we have now is too much code duplication for invoking commands with powershell instance.

Reply:
Initially you proposed that the method is only for completers but here is another code.
The method has 3 parameters but implemented with condition to skip last one. Bad design.
I found ~120 `PowerShell.Create` in our code but code there is very simple, I don't see cases where we need a filtering.

Maybe I missed something, but now I think that if you do a generic helper method, it's only for completers and only with filtering. In other cases, it unnecessarily complicates the code.

Reply:
> Initially you proposed that the method is only for completers but here is another code.

Incorrect. This code is being used inside the `PropertyType` completer. There is no code I am changing which is not for completers.

> The method has 3 parameters but implemented with condition to skip last one. Bad design.

I don't understand this. There is a case where we pass in `fakeBoundParameters` directly and do filtering, and then there is a case where we pass in parameters directly. The method supports both. 

> I found ~120 PowerShell.Create in our code but code there is very simple, I don't see cases where we need a filtering.

Yes but majority of these references is not for completers. Completion code usually requires some level of filtering with `fakeBoundParameters`.

Reply:
Thanks for clarify!

> Completion code usually requires some level of filtering with fakeBoundParameters.

Can you share other completers? Maybe include them in the PR? Are there examples others than Get-Command and Resolve-Path we could run with the helper method?

Reply:
https://github.com/PowerShell/PowerShell/blob/14fb6f26183cc495f28a6a68a9f2fdefa91b0dd6/src/System.Management.Automation/engine/ExperimentalFeature/EnableDisableExperimentalFeatureCommand.cs#L122-L125

Here is one I was going to prepare to refactor by doing this PR first.

But I am more than happy to leave this if design is questionable, I don't see other options to generalize this without doing the same duplication. 

Its a matter of preference, I prefer it but if it not adding much value, I am more than happy to move on and just refactor the `ExperimentalFeatureNameCompleter` completer to support quotes 🙂.

Reply:
It definitely doesn't add to the readability of the code, but if it's going to be used frequently (in every second completer), then it's worth it, otherwise I wouldn't create this entity.